packages/client-config/API.md

```diff
@@ -207,10 +210,25 @@ export type CustomClientConfig = {
 export const DEFAULT_CLIENT_CONFIG_VERSION: ClientConfigVersion;
 
 // @public
-export const generateClientConfig: <T extends "1" | "0">(credentialProvider: AwsCredentialIdentityProvider, backendIdentifier: DeployedBackendIdentifier, version: T) => Promise<ClientConfigVersionTemplateType<T>>;
+export const generateClientConfig: <T extends "1" | "0">(awsClientProvider: AWSClientProvider<{
+    getS3Client: S3Client;
+    getAmplifyClient: AmplifyClient;
+    getCloudFormationClient: CloudFormationClient;
+}>, backendIdentifier: DeployedBackendIdentifier, version: T) => Promise<ClientConfigVersionTemplateType<T>>;
+
+// @public (undocumented)
+export type GenerateClientConfigOptions = AWSClientProvider<{
+    s3Client: S3Client;
+    cloudFormationClient: CloudFormationClient;
+    amplifyClient: AmplifyClient;
+}>;

```

Comment:
if we do `GenerateClientConfigOptions` I suggest we do:

```
type GenerateClientConfigOptions = {
   awsClientProvider: AWSClientProvider
}
```

for future expansion. (i.e. future proof it, so we don't have to do MV bumps.).

And if we do this we should rename parameter to `generateCLientCofnig` to options.

Reply:
Actually, I removed this type. Since only AWSClientProvider is needed we can just use that. Do you think it's better as a bag of options with just this one option for now?

Reply:
> Do you think it's better as a bag of options with just this one option for now?

I'm not 100% sure of the future of this API.
If we get more options then having options bag now is better because we avoid breaking change (or blending provider interface with other options).
However, shipping major rev of this particular package later seems ok to me.
@edwardfoyle @Amplifiyer what do you think ?

Reply:
In my opinion, shipping major rev is actually the right thing when we start adding new parameters to APIs instead of keeping a bag that anyone can put anything later on. 

Moreover, the optional arguments can still be added both ways without breaking any clients.